Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -926,36 +948,30 @@ DirectoryBlobPartition::DirectoryBlobPartition(
   if (fst_callback)
     fst_callback(&nodes, &dol_node);
 
-  const u64 new_fst_address = SetDOL(std::move(dol_node), new_dol_address);
-  BuildFST(std::move(nodes), new_fst_address);
-}
+  const u64 new_fst_address = SetDOL(std::move(dol_node), new_dol_address, &disc_header);
+  BuildFST(std::move(nodes), new_fst_address, &disc_header);
 
-void DirectoryBlobPartition::SetDiscHeaderFromFile(const std::string& boot_bin_path)
-{
-  m_disc_header.resize(DISCHEADER_SIZE);
-  if (ReadFileToVector(boot_bin_path, &m_disc_header) < 0x20)
-    ERROR_LOG_FMT(DISCIO, "{} doesn't exist or is too small", boot_bin_path);
-
-  m_contents.AddReference(DISCHEADER_ADDRESS, m_disc_header);
+  m_contents.Add(DISCHEADER_ADDRESS, disc_header);
 }
 
-void DirectoryBlobPartition::SetDiscHeader(std::vector<u8> boot_bin)
+DirectoryBlobPartition::DirectoryBlobPartition(const DirectoryBlobPartition& rhs)
+    : m_contents(rhs.m_contents), m_key(rhs.m_key), m_root_directory(rhs.m_root_directory),
+      m_is_wii(rhs.m_is_wii), m_address_shift(rhs.m_address_shift), m_data_size(rhs.m_data_size),
+      m_wrapped_partition(rhs.m_wrapped_partition)
```

Comment:
This copy constructor looks 100% standard. You could replace the implementation with `=  default`.